astroquery/esa/hubble/core.py

```diff
@@ -541,6 +540,43 @@ def query_target(self, name, *, filename=None, output_format='votable',
 
         return table
 
+    def query_tap(self, query, async_job=False, output_file=None,
```

Comment:
This may better be a `query_tap_async` than driven with the `async_job` kwarg, and the whole class decorated with `async_to_sync`, but I would not add that into this PR but wait for the discussion in https://github.com/astropy/astroquery/issues/2598. 

Input in that issue is more than welcome

Reply:
The problem we had here is that in TapPlus class we are using launch_job_sync and launch_job_async. It is not a matter of modifying it on the mission classes. What we should do is check how it is working in the common class, modify it there and extend the changes to ESA modules. This has to be modified carefully... I agree, maybe we can do this in a different pull request.

Reply:
the discussion is geared towards dropping the decorator approach in favour of the kwarg, so no extensive modification is expected.